Electron/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -1,85 +1,273 @@
-import { DATAID } from "./RovecommManifest"
+import { Socket } from "dgram"
+import { Server } from "http"
+import { DATAID, dataSizes, DataTypes } from "./RovecommManifest"
+
+// There is a fundamental implementation difference between these required imports
+// and the traditional typescript imports.
 /* eslint-disable @typescript-eslint/no-var-requires */
 const dgram = require("dgram")
 const net = require("net")
 const EventEmitter = require("events")
 
-function TCPListen(socket: any) {
-  socket.on("data", (data: any) => {
-    console.log(data, data.toString())
+function decodePacket(
+  dataType: number,
+  dataLength: number,
+  data: Buffer
+): number[] {
+  /*
+   * Takes in a dataType, dateLength, and data from an incoming rovecomm packet,
+   * and uses the dataType to return an array of the properly typed data.
+   * Note: even if dataLength is only 1, this returns an array of one item.
+   */
+
+  let readBytes: (i: number) => number
+
+  switch (dataType) {
+    case DataTypes.INT8_T:
+      readBytes = data.readInt8.bind(data)
+      break
+    case DataTypes.UINT8_T:
+      readBytes = data.readUInt8.bind(data)
+      break
+    case DataTypes.INT16_T:
+      readBytes = data.readInt16BE.bind(data)
+      break
+    case DataTypes.UINT16_T:
+      readBytes = data.readUInt16BE.bind(data)
+      break
+    case DataTypes.INT32_T:
+      readBytes = data.readInt32BE.bind(data)
+      break
+    case DataTypes.UINT32_T:
+      readBytes = data.readUInt32BE.bind(data)
+      break
+    case DataTypes.FLOAT_T:
+      readBytes = data.readFloatBE.bind(data)
+      break
+    default:
+      return []
+  }
+
+  const retArray = []
+  let offset: number
+  for (let i = 0; i < dataLength; i += 1) {
+    offset = i * dataSizes[dataType]
+    retArray.push(readBytes(offset))
+  }
+  return retArray
+}
+
+function parse(packet: Buffer): void {
+  /*
+   * Parse takes in a packet buffer and will call decodePacket and emit
+   *  the rovecomm event with the proper dataId and that typed data
+   *
+   *  RoveComm Header Format:
+   *
+   *   0                   1                   2                   3
+   *   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   *  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   *  |    Version    |            Data Id            |  Data Length  |
```

Comment:
dataLength may be a little ambiguous, it's technically the data count. Length might imply that its the size in bytes, while it's the number of seperate data entries. Not a big deal either way.

Reply:
Yeah, we should probably switch this to Data Count in the two current instances to ensure easier readability for future maintainers.